### PR TITLE
feat(protocol-designer): add redux infra for batch edit mode

### PR DESCRIPTION
**Pull request by shlokamin**
# Overview

@IanLondon  as discussed, this PR splits up my WIP PR (#7122) to just include the redux infra so it can be used for other tickets.

# Changelog
- Add batch edit redux infra

# Review requests
Code review, nothing is actually plugged in yet.

# Risk assessment
Low/Med, touches redux but nothing is actually being used in this PR (multi select actions not being dispatched, nobody using this part of the store)

Also, lot of tip-related stuff that wasn't in use was deleted!

### CODE REVIEW

File: protocol-designer/src/components/DeckSetup/LabwareOnDeck.js [hunk 1]
```diff
@@ -43,16 +43,24 @@ const LabwareOnDeckComponent = (props: Props) => (
...
+    state
+  )
+
   return {
-    wellContents: wellContentsSelectors.getAllWellContentsForActiveItem(state)[
-      labwareOnDeck.id
-    ],
+    wellContents: allWellContentsForActiveItem
+      ? allWellContentsForActiveItem[labwareOnDeck.id]
+      : {}, // TODO(IL, 2020-12-15): null instead of {}? See #7135
```
Comment (shlokamin) [inline protocol-designer/src/components/DeckSetup/LabwareOnDeck.js:57]: did we make a call on `null` vs `{}`?

File: protocol-designer/src/components/DeckSetup/LabwareOnDeck.js [hunk 2]
```diff
@@ -43,16 +43,24 @@ const LabwareOnDeckComponent = (props: Props) => (
...
     highlightedWells: highlightSelectors.wellHighlightsByLabwareId(state)[
       labwareOnDeck.id
     ],
-    missingTips: tipContentsSelectors.getMissingTipsByLabwareId(state)[
-      labwareOnDeck.id
-    ],
+    missingTips: missingTipsByLabwareId
+      ? missingTipsByLabwareId[labwareOnDeck.id]
+      : {}, // TODO(IL, 2020-12-15): null instead of {}? See #7135
```
Comment (shlokamin) [inline protocol-designer/src/components/DeckSetup/LabwareOnDeck.js:63]: same question as above

File: protocol-designer/src/components/labware/BrowseLabwareModal.js [hunk 3]
```diff
@@ -66,9 +66,14 @@ function mapStateToProps(state: BaseState): SP {
...
-    ? wellContentsSelectors.getAllWellContentsForActiveItem(state)[labwareId]
-    : {}
+
+  const allWellContentsForActiveItem = wellContentsSelectors.getAllWellContentsForActiveItem(
+    state
+  )
+  const wellContents =
+    labwareId && allWellContentsForActiveItem
+      ? allWellContentsForActiveItem[labwareId]
+      : {} // TODO(IL, 2020-12-15): null instead of {}? See #7135
```
Comment (shlokamin) [inline protocol-designer/src/components/labware/BrowseLabwareModal.js:76]: same here

File: protocol-designer/src/top-selectors/tip-contents/index.js [hunk 4]
```diff
@@ -1,108 +1,22 @@
...
-)
-
-const getLastValidTips: GetTipSelector = createSelector(
-  fileDataSelectors.lastValidRobotState,
-  getLabwareIdProp,
-  (lastValidRobotState, labwareId) => (wellName: string) => ({
-    empty: getTipEmpty(wellName, labwareId, lastValidRobotState),
-    highlighted: false,
-  })
-)
```
Comment (shlokamin) [inline protocol-designer/src/top-selectors/tip-contents/index.js:101]: 😄

**Review by shlokamin — COMMENTED** [review]
I can't approve since i technically opened this, but it looks good! I smoke tested and nothing seems to be broken. Lets get this in!

File: protocol-designer/src/components/DeckSetup/LabwareOnDeck.js (hunk 1 above)
Comment (IanLondon) [reply protocol-designer/src/components/DeckSetup/LabwareOnDeck.js:57]: I like `null` better but it requires changes that would make this PR pretty big so I broke that off into #7135

Though if there's a case to keep it `{}` I'm down to hear it

**Review by IanLondon — APPROVED** [review]

**Review by Kadee80 — APPROVED** [review]
let's do it! 🔥

**Pull request merged**